tools/tests/scrollable-container/dynamic-content.test.js

```diff
@@ -0,0 +1,119 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Coherent Labs AD. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+function showDynamicScrollbarContent(scrollableContainer) {
+    scrollableContainer.querySelector('.dynamic-content').style.display = 'block';
+}
+
+function hideDynamicScrollbarContent(scrollableContainer) {
+    scrollableContainer.querySelector('.dynamic-content').style.display = 'none';
+}
+
+describe('Scrollable Container Component', () => {
+    afterAll(() => cleanTestPage('.scrollable-container-test-wrapper'));
+
+    beforeEach(function (done) {
+        const template = `<gameface-scrollable-container class="scrollable-container" automatic>
+        <component-slot data-name="scrollable-content">
+        Some short content.
+        <div class="dynamic-content" style="display:none;">${dynamicContent}</div>
+        </component-slot>
+        </gameface-scrollable-container>`;
+
+        const el = document.createElement('div');
+        el.innerHTML = template;
+        el.className = 'scrollable-container-test-wrapper';
+
+        cleanTestPage('.scrollable-container-test-wrapper');
+
+        document.body.insertBefore(el, document.body.firstElementChild);
+
+        waitForStyles(() => {
+            done();
+        })
+    });
+
+    it('Should be rendered', () => {
+        assert(document.querySelector('.scrollable-container') !== null, 'Scrollable container is not rendered.');
+    });
+
+    it('Should not show scrollbar as the content does not overflow', () => {
+        const style = getComputedStyle(document.querySelector('.slider-component'));
+
+        return createAsyncSpec(() => {
+            assert(style.display === 'none', 'The scrollbar is visible.');
+        });
+    });
+
+    it('Should show scrollbar if the content is changed and overflows', async () => {
+        const scrollableContainer = document.querySelector('.scrollable-container');
+        showDynamicScrollbarContent(scrollableContainer);
+
+        return createAsyncSpec(() => {
+            const sliderStyle = getComputedStyle(scrollableContainer.querySelector('.slider-component'));

```

Comment:
Should we wait when getting getComputedStyle like it is in the test above?

Reply:
In the previous test we wait to make sure the initial style computation when the scrollbar is attached has completed. In this one we change the style, wait for it to be re-computed and get it. The actual accessing of the style doesn't require waiting, its computation does. So in the second case - no, we don't need to wait.